src/howler.core.js

```diff
@@ -1484,6 +1535,9 @@
           // Remove any event listeners.
           sounds[i]._node.removeEventListener('error', sounds[i]._errorFn, false);
           sounds[i]._node.removeEventListener(Howler._canPlayEvent, sounds[i]._loadFn, false);
+
+          // Release the Audio object back to the pool.
+          Howler._releaseHtml5Audio(sounds[i]._node);
```

Comment:
Yes, the pool can get muddied. Absent a means of identifying an activated Audio object (via an extra property, for example), the ideal is to ensure that the pool is configured in such a fashion as to ensure it never runs out. In a properly sized world, this will never happen.